### ISSUE TITLE
Allow scripts to receive user input during runtime

### ISSUE THREAD

**Issue by caleb531** [rgbkrk/atom-script#280]
Hi,

Currently, Atom Script works very well for quickly checking the output of a program within the editor. However, programs may often require user input, and it seems that Atom Script does not allow for the user to type input and send it to a program when the program requests it.

For instance, the following Python 2 code never finishes but instead hangs, waiting for the user to provide input which cannot be typed anywhere.

```python
name = raw_input('What is your name?')
print 'Hello, ' + name + '.'
```

Therefore, I think the ability to pass user input to a program on runtime is a well-needed feature. Any possibility of adding this?

Thanks,
Caleb

**Comment by Edenhofer** [comment]
+1

**Comment by rgbkrk** [comment]
I'd sure like to see that too.

**Comment by myrddin89** [comment]
Hi,
I would really appreciate this feature too.

**Comment by max-santiago** [comment]
I really need this feature.

**Comment by rgbkrk** [comment]
Check out [Hydrogen](https://github.com/willwhitney/hydrogen) if you'd like an advanced REPL embedded in Atom.

**Comment by etem** [comment]
+1

**Comment by phoenixyu** [comment]
I need this feature :)

**Comment by himrasmussen** [comment]
Ditto

**Comment by YugoCode** [comment]
Any updates?

**Comment by nik4mub** [comment]
Essential feature!

**Comment by NonlinearFruit** [comment]
This would help set Atom apart from Sublime

**Comment by petersteele111** [comment]
Yeah just ran into this issue as well, would be really handy not having to jump into another idle or CLI to do it

**Comment by jdleo** [comment]
Just came across this thread. Would LOVE to see this.

**Comment by timecode** [comment]
Me too. Would be a really helpful addition to be able to receive user input during runtime.

**Comment by jdleo** [comment]
@timecode yeah, kind of tedious to run the file thru terminal every single time you make a change if your code has user input :|

**Comment by Ahmedbayoumi** [comment]
Sure, just was searching for a solution

**Comment by psmitsu** [comment]
Is it possible to specify input and output files in Atom script? I have tried to put "< input.txt" in Command arguments, but when I do cmd+i it says "bash: <: No such file or directory".

**Comment by boblu** [comment]
Yes. Accept user's input will be really handy here. +1 for this.

**Comment by Neviothr** [comment]
I'd also like this.

**Comment by saikpr** [comment]
Hey there is a sublime text package https://github.com/mavidser/SublimeInput

can we use the python script they have used there to parse?

**Comment by antony5386** [comment]
I need this too!

**Comment by lgeiger** [comment]
@antony5386 As mentioned above you can try [Hydrogen](https://github.com/nteract/hydrogen) which supports STDIN.

**Comment by antony5386** [comment]
@lgeiger  I didn't see it the first time. thanks a lot!

**Comment by ilhamahmad** [comment]
i need this

**Comment by wellington1993** [comment]
I need this too.

**Comment by ktan18879** [comment]
This is a really important consideration. Currently is there any workaround?

**Comment by habi307** [comment]
I'm facing the issue discussing here ,Kindly tell me what should i do ?what should I do to take user input?

**Comment by Adrian-Voelker** [comment]
What is about that feature?

**Comment by natonboy** [comment]
seems like no progress... uninstalling it.
hydrogen seems great, but input() always darken my screen and outputs too far below out of my screen.

**Comment by DavidAMills** [comment]
I'm surprised that after four years there is still no STDIN capability in Scripts, but Hydrogen was a good suggestion, works great!
